### PR TITLE
[CdapIO] Fixed necessary warnings

### PR DESCRIPTION
**Pull request by akashorabek**
In CdapIO we have some warnings that come out when trying to build a project. Previously, we suppressed these warnings with the @SuppressWarnings annotation.

The purpose of this PR is to solve the necessary warnings without the @SuppressWarnings annotation and loss of IO functionality.

### CODE REVIEW

File: sdks/java/io/cdap/src/test/java/org/apache/beam/sdk/io/cdap/CdapIOTest.java [hunk 1]
```diff
@@ -118,12 +118,6 @@ public void testReadObjectCreationFailsIfValueClassIsNull() {
         IllegalArgumentException.class, () -> CdapIO.<String, String>read().withValueClass(null));
   }
 
-  @Test
-  public void testReadValidationFailsMissingCdapPluginClass() {
```
Comment (Abacn) [inline sdks/java/io/cdap/src/test/java/org/apache/beam/sdk/io/cdap/CdapIOTest.java:122]: Good to preserve these error handling tests. now it throws at expansion.

**Review by Abacn — CHANGES_REQUESTED** [review]
Thanks. This removed two of them warning suppressions in CdapIO. There are also other suppression annotations exists currently, are there also considered need to be resolved?

File: sdks/java/io/cdap/src/test/java/org/apache/beam/sdk/io/cdap/CdapIOTest.java (hunk 1 above)
Comment (akashorabek) [reply sdks/java/io/cdap/src/test/java/org/apache/beam/sdk/io/cdap/CdapIOTest.java:122]: Done

**Comment by Lizzfox** [comment]
> There are also other suppression annotations exists currently, are there also considered need to be resolved?

@Abacn, thank you for the comment. Yes, there are two classes with warnings suppression left in the CdapIO package. 
1. In the `Plugin` class, the warnings will be fixed in the following PR where we will integrate SparkReceiverIO with CdapIO.

2. The `BatchContextImpl` class is an abstraction that we've implemented to make it possible to use CDAP abstractions in the Apache Beam context. Methods that produce warnings are overridden so we can't change their signature to resolve warnings. The suppressions in the `BatchContextImp` class aren't expected to be resolved anytime soon.

**Review by Abacn — APPROVED** [review]
Thanks. LGTM

R: @chamikaramj

**Pull request merged**